packages/merge-styles/src/mergeStyleSets.ts

```diff
@@ -11,17 +11,30 @@ import { IStyle } from './IStyle';
 export function mergeStyleSets<T>(
   ...cssSets: ({[P in keyof T]?: IStyle } | null | undefined)[]
 ): T {
-  const classNameSet: Partial<T> = {};
+  // tslint:disable-next-line:no-any
+  const classNameSet: any = {};
```

Comment:
Can you not use `{[P in keyof T]?: string}` instead of `any`?

Reply:
I have tried a lot of things. No can't do that.

```
Argument of type '{ [P in keyof T]?: string | undefined; }' is not assignable to parameter of type '{ [key: string]: string; } | undefined'.
  Type '{ [P in keyof T]?: string | undefined; }' is not assignable to type '{ [key: string]: string; }'.
    Type 'string | undefined' is not assignable to type 'string'.
      Type 'undefined' is not assignable to type 'string'.
const classNameSet: {
    [P in keyof T]?: string | undefined;
}
```

Reply:
I can't remove the ? either without making something else unhappy.